x-pack/plugins/ingest_manager/server/services/epm/fields/field.ts

```diff
@@ -108,18 +110,54 @@ function dedupFields(fields: Fields): Fields {
       return f.name === field.name;
     });
     if (found) {
+      // remove name, type, and fields from `field` variable so we avoid merging them into `found`
+      const { name, type, fields: nestedFields, ...importantFieldProps } = field;
+      /**
+       * There are a couple scenarios this if is trying to account for:
+       * Example 1
+       *  - name: a.b
+       *  - name: a
+       *  In this scenario found will be `group` and field could be either `object` or `nested`
+       * Example 2
+       *  - name: a
+       *  - name: a.b
+       *  In this scenario found could be `object` or `nested` and field will be group
+       */
       if (
-        (found.type === 'group' || found.type === 'object') &&
-        field.type === 'group' &&
-        field.fields
+        // only merge if found is a group and field is object, nested, or group.
+        // Or if found is object, or nested, and field is a group.
+        // This is to avoid merging two objects, or nested, or object with a nested.
+        (found.type === 'group' &&
```

Comment:
@skh @ruflin let me know if you have suggestions for cleaning this logic up.

Reply:
For now, I'm fine with it, especially with all the comments you added.

I'd suggest adding an issue in the kibana repo for beta1 to revisit `dedupFields()` after alpha1, fully specify it's behavior (see also https://github.com/elastic/package-registry/issues/340 ) and verify that it handles all edge cases correctly.